### ISSUE TITLE
Java sims not launching through JNLP

### ISSUE THREAD

**Issue by jbphet** [phetsims/website#472]
After the switch to the new server, it appears that Java sims are not launching via the JNLP links.  On my Win 10 machine, I see the following dialog:

![jnlp-err](https://cloud.githubusercontent.com/assets/3082831/17597980/875fb2fa-5fb5-11e6-8883-cd20e8ff88dd.PNG)

**Comment by oliver-phet** [comment]
MP is going to write an exception so that .jnlp paths aren’t redirected to https

**Comment by jbphet** [comment]
After looking at the exceptions - specifically the part that mentions an "illegal redirect" - it looks like the issue is probably that the JNLP references an http location and the website currently redirects everything to https.

Below are screenshots of the exceptions, which can be seen when the user presses the "Details" button in the dialog above:

![jnlp-excp1](https://cloud.githubusercontent.com/assets/3082831/17598790/6938d456-5fb9-11e6-8e13-a17bf69389f0.PNG)

![jnlp-excp2](https://cloud.githubusercontent.com/assets/3082831/17598796/6f75b186-5fb9-11e6-8e03-ffb2dd881a29.PNG)

**Comment by jbphet** [comment]
Assigning to @mattpen to set up an exception so that the references to the JNLP files aren't redirected.

**Comment by mattpen** [comment]
We have added an exception that allows the .jnlp files over http, but the jnlp launches are now failing for the same reason for the .jar file itself.  We'll add .jar files to the SSL redirect exception too.

**Comment by jbphet** [comment]
Over Skype, @jonathanolson said:
 >[S]ome other places probably have JNLPs that we can't change that ref http. Http exception sounds good for now.

**Comment by mattpen** [comment]
I added a rule to allow http access to the entire /sims directory, and the sims are now launching correctly.

**Comment by mattpen** [comment]
> [S]ome other places probably have JNLPs that we can't change that ref http. Http exception sounds good for now.

Just my two cents .... It might not be a bad idea to go ahead and change the jnlp files so that they point https now, at least if the development effort required is not preventative.  Even if we leave the http exception in place for the foreseeable future, we may eventually want to require https for the entire site.

**Comment by kathy-phet** [comment]
I agree with this, especially if it development effort is minimal -- e.g. run a script over the posted .jnlp files and change http to https?

**Comment by jbphet** [comment]
Changing it on the existing .jnlp files is part of the solution, but we should keep in mind that the build tools for the Java sims, which are still in use for publishing translations, will also need to be updated to generate the correct links in the .jnlp files.  Probably not a huge deal, but we need to be sure that we don't neglect this.  We can potentially do it as part of the effort to make these tools work for the new server, see https://github.com/phetsims/tasks/issues/672.

**Comment by jbphet** [comment]
Also, if we make the changes to the .jnlp files, we will need to look at the installer builder and see if it needs to be changed too, since it interprets and modifies the .jnlp files as part of the process of creating the installers.

**Comment by mattpen** [comment]
Do we want to enforce https on the `/sims/html` directory?

**Comment by kathy-phet** [comment]
Will it redirect automatically if folks type http? If so, then I think yes, unless there is a downside I don't understand.

Some companies have said they require us to work with https.

Sent from my iPhone

On Aug 11, 2016, at 10:14 PM, Matt Pennington <notifications@github.com<mailto:notifications@github.com>> wrote:


Do we want to enforce https on the /sims/html directory?

-

**Comment by jbphet** [comment]
Marking for discussion in developer meeting to decide whether to make the java sims launch from https and, if so, who will do the work and when.

**Comment by jbphet** [comment]
This was discussed at the 8/18/2016 meeting and it was decided that @mattpen will do a prelim investigation and provide an estimate but we will probably wait until @jonathanolson is available before taking action on this.

**Comment by mattpen** [comment]
@ariel-phet - Can you please assign a priority? Should @jonathanolson and I work on this soon?

**Comment by mattpen** [comment]
This should ideally be tackled alongside #672

**Comment by ariel-phet** [comment]
@mattpen yes this would be good to work on in the next month. Fixing the build tools also makes sense since it is at least possible some action will need to be taken once we have our new certificate.

**Comment by mattpen** [comment]
@jonathanolson - Are you available to pair on this task, sometime in the next 2 weeks?

**Comment by jonathanolson** [comment]
Yes, should be available, let me know when is convenient.

**Comment by zepumph** [comment]
When you guys do this, could I be involved too, I'd love to get a little more understanding on this part of the project.

**Comment by ariel-phet** [comment]
Adding @zepumph to assignee list since he is expressing interest

**Comment by zepumph** [comment]
Unassigning as I am now quite busy. Let me know if you need anything.
